scripts/regenerate-lock-files

```diff
@@ -85,28 +94,35 @@ def delete_file(path: Path):
         pass
 
 
-def main():
+def main(base_directory: str):
     build_lock_file(
         sources=[os.path.join("download-deps", "bootstrap.txt")],
-        output=os.path.join("download-deps", "bootstrap"),
+        output=os.path.join(base_directory, "download-deps", "bootstrap"),
```

Comment:
It seems off on how we are using the `output` parameter. Based on the code, it seems it will be always rooted in the source directory because the `_output_path()` function always uses `REQ_DIR` as the root. Instead it seems we should be making the `base_directory` the root. From there we should be able to get around the Windows `relpath` issues by calling `os.path.relpath()` on the full output path and the output dir.

Reply:
It's not rooted in the source directory if you want to generate the output elsewhere, like in the test.